spinalcordtoolbox/deepgmseg/deepgmseg.py

```diff
@@ -0,0 +1,215 @@
+# coding: utf-8
+
+import json
+import os
+
+import nipy
+import nibabel as nib
+from nipy.io.nifti_ref import nipy2nifti, nifti2nipy
+import numpy as np
+
+
+from spinalcordtoolbox.resample import nipy_resample
+from . import model
+
+
+class DataResource(object):
+    def __init__(self, dirname):
+        """Initialize the resource with the directory
+        name context.
+
+        :param dirname: the root directory name.
+        """
+        path_script = os.path.dirname(__file__)
+        data_dir = os.path.join(path_script,
+                                "..", "..", "data")
+        data_dir = os.path.abspath(data_dir)
+        self.data_root = os.path.join(data_dir, dirname)
+
+    def get_file_path(self, filename):
+        """Get the absolute file path based on the
+        data root directory.
+
+        :param filename: the filename.
+        """
+        return os.path.join(self.data_root, filename)
+
+
+class CroppedRegion(object):
+    """This class holds cropping information about the volume
```

Comment:
"volume". But later original_shape is 2D?

Reply:
This is a matter of point of view, the fact that the data structure is 2D doesn't mean that it isn't a volume because we still have voxels with 3D dimensions in this 2D data. I prefer to leave as that to enforce the semantic meaning of this data.